esmvaltool/diag_scripts/hydrology/lazy_regrid.py

```diff
@@ -46,58 +44,17 @@ def _compute_chunks(src, tgt):
         (src.coord('latitude').shape[0], ),
         (src.coord('longitude').shape[0], ),
     )
-    tgt_chunks = (
-        time_chunks,
-        (tgt_nlat, ),
-        (tgt_nlon, ),
-    )
 
-    return src_chunks, tgt_chunks
+    return src_chunks
 
 
-def _regrid_data(src, tgt, scheme):
-    """Regrid data from cube src onto grid of cube tgt."""
-    src_chunks, tgt_chunks = _compute_chunks(src, tgt)
+def lazy_regrid(src, tgt, scheme):
+    """Regrid cube src onto the grid of cube tgt."""
+    src_chunks = _compute_chunks(src, tgt)
 
-    # Define the block regrid function
     if scheme not in HORIZONTAL_SCHEMES:
         raise ValueError(f"Regridding scheme {scheme} not supported, "
                          f"choose from {HORIZONTAL_SCHEMES.keys()}.")
     regridder = HORIZONTAL_SCHEMES[scheme].regridder(src, tgt)
-
-    def regrid(block):
-        tlen = block.shape[0]
-        cube = src[:tlen].copy(block)
-        return regridder(cube).core_data()
-
-    # Regrid
-    data = src.core_data().rechunk(src_chunks).map_blocks(
-        regrid,
-        dtype=src.dtype,
-        chunks=tgt_chunks,
-    )
-
-    return data
-
-
-def lazy_regrid(src, tgt, scheme):
-    """Regrid cube src onto the grid of cube tgt."""
-    data = _regrid_data(src, tgt, scheme)
-
-    result = iris.cube.Cube(data)
-    result.metadata = copy.deepcopy(src.metadata)
-
-    def copy_coords(src_coords, add_method):
-        for coord in src_coords:
-            dims = src.coord_dims(coord)
-            if coord == src.coord('longitude'):
-                coord = tgt.coord('longitude')
-            elif coord == src.coord('latitude'):
-                coord = tgt.coord('latitude')
-            result_coord = coord.copy()
-            add_method(result_coord, dims)
-
-    copy_coords(src.dim_coords, result.add_dim_coord)
-    copy_coords(src.aux_coords, result.add_aux_coord)
-
-    return result
+    src.data = src.lazy_data().rechunk(src_chunks)
+    return regridder(src)
```

Comment:
This can just be replaced with
```python
src.data = src.lazy_data().rechunk(src_chunks)
return esmvalcore.preprocessor.regrid(src, tgt, scheme)
```
and maybe you could rename the function to something like `low_memory_regrid` or `rechunk_and_regrid` or something? Because lazy doesn't really describe what the function does anymore, the usual `esmvalcore.preprocessor.regrid` function is also lazy.

Reply:
Thank you, it is fixed as you suggested, iris is replaced with esmvalcore. The function is also renamed to `rechunk_and_regrid` now. Shall we rename the script name too? any suggestion?

Reply:
If wflow is the only place we use it, we might as well call the script "rechunk.py", and move the few remaining lines that do the regridding back to the wflow diagnostic. What do you think?

Reply:
GlobWat (in the draft PR #1808) uses this script too.

Reply:
> any suggestion?

It's up to you

Reply:
I implemented @Peter9192 suggestions about moving the regridding codes to the diagnostic. I renamed the script to `compute_chunks.py.`